packages/launchpad/src/signer.ts

```diff
@@ -19,7 +28,25 @@ export interface SignResponse {
   readonly signature: StdSignature;
 }
 
-export interface OfflineSigner {
+export interface DirectSignResponse {
+  readonly signed: Uint8Array;
+  readonly signature: StdSignature;
+}
+
+export interface AminoSignSpec {
+  readonly signJob: AminoSignJob;
+  readonly signResponse: AminoSignResponse;
+}
+
+export interface DirectSignSpec {
+  readonly signJob: DirectSignJob;
+  readonly signResponse: DirectSignResponse;
+}
+
+export type SignSpec = AminoSignSpec | DirectSignSpec;
+
+export interface OfflineSigner<T extends SignSpec> {
```

Comment:
After a bit more reflection, and reading through the code, I think the most straight-forward approach uses more simple types:

```typescript
export interface OfflineAminoSigner {
  readonly getAccounts: () => Promise<readonly AccountData[]>;
  readonly signAmino: (signerAddress: string, signJob: AminoSignJob) => Promise<AminoSignResponse>;
}

export interface OfflineAminoDirect {
  readonly getAccounts: () => Promise<readonly AccountData[]>;
  readonly signDirect: (signerAddress: string, signJob: DirectSignJob) => Promise<DirectSignResponse>;
} 
```

You can add some `isOfflineAminoSigner` and `isOfflineDirectSigner` helpers to detect this at runtime. And a wallet can easily implement both.

You heavily rely on compile-time type info here, which is not present at runtime. When we deal with eg. injected providers from Keplr, how can you determine what support it provides? Also, how to support both in one signer? Ideally a software Secp256k1Wallet could support both and just drop in to any signer - that is not really possible with the current system.